hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/InlineDirtyCheckingHandler.java

```diff
@@ -96,6 +101,93 @@ public Size apply(
 			MethodVisitor methodVisitor,
 			Context implementationContext,
 			MethodDescription instrumentedMethod) {
+
```

Comment:
just wondering, but was there a specific reason to have preferred using an ASM visitor over using a ByteBuddy template? Could you add a comment about the decision, for sake of future maintainers?

Reply:
No particular reason. I've just been following the already existing pattern in that class.